coalib/testing/LocalBearTestHelper.py

```diff
@@ -52,6 +54,22 @@ class LocalBearTestHelper(unittest.TestCase):
     If you miss some methods, get in contact with us, we'll be happy to help!
     """
 
+    def assertComparableObjectsEqual(self, observed_result, expected_result):
+        self.assertEqual(len(observed_result), len(expected_result),
+                         msg='The number of elements in the observed result '
+                             'and expected result differ.')
```

Comment:
No, your proposed code here still only looks at length.

Checking length does not do "check if the length difference is due to only new items or only deleted items."

To determine if it is only new or only deleted items, you *must* look at each expected item and find it in the resulting array.  Then you can report exactly which item(s) was new or deleted.  That is what assertEqual already does.